tests/AbstractFactoryDecoratorTest.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+namespace byrokrat\id;
+
+use PHPUnit\Framework\TestCase;
+
+class AbstractFactoryDecoratorTest extends TestCase
+{
+    public function testConstruct()
+    {
+        $factory = new class extends AbstractFactoryDecorator {
+            public function createNewInstance(?string $raw): IdInterface {}
+        };
+
+        $this->assertInstanceOf(IdFactory::class, $factory);
+    }
+
+    public function testCreateId()
+    {
+        $factory = new class extends AbstractFactoryDecorator {
```

Comment:
One could use `getMockForAbstractClass` and return a mock of `IdInterface`. See comment below..

Reply:
In that case we would assert only on mocks.